frontend/src/components/Signup/Signup.js

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from "react"
+import basestyle from "../Base.module.css"
+import registerstyle from "./Signup.module.css"
+import axios from "axios"
+
+import { useNavigate, NavLink } from "react-router-dom"
+const Signup = () => {
+  const navigate = useNavigate()
+
+  const [formErrors, setFormErrors] = useState({})
+  const [isSubmit, setIsSubmit] = useState(false)
+  const [user, setUserDetails] = useState({
+    name: "",
+    phone: "",
+    email: "",
+    password: "",
+    confirmpwd: "",
+  })
+  const [error, setError] = useState("")
+
+  const changeHandler = (e) => {
+    const { name, value } = e.target
+    setUserDetails({
+      ...user,
+      [name]: value,
+    })
+  }
+
+  const validateForm = (values) => {
+    const error = {}
+    const matchPattern = /^[^\s+@]+@[^\s@]+\.[^\s@]{2,}$/i
```

Comment:
Can you change it as emailRegex or something related to email. Match pattern is generous.